test/e2e/autotls/domain_mapping_test.go

```diff
@@ -0,0 +1,135 @@
+	"knative.dev/networking/test/conformance/ingress"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/reconciler"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+	"knative.dev/serving/pkg/apis/serving/v1alpha1"
+	"knative.dev/serving/test"
+	"knative.dev/serving/test/e2e"
+	v1test "knative.dev/serving/test/v1"
+)
+
+type dmConfig struct {
+	// ServiceName is the name of testing Knative Service.
+	// It is not required for self-signed CA or for the HTTP01 challenge when wildcard domain
+	// is mapped to the Ingress IP.
+	TLSServiceName string `envconfig:"tls_service_name" required:"false"`
+	// CustomDomainSuffix is the custom domain used for the domainMapping
+	CustomDomainSuffix string `envconfig:"custom_domain_suffix" required:"false"`
+}
+
+func TestDomainMappingAutoTLS(t *testing.T) {
+	var env dmConfig
+	if err := envconfig.Process("", &env); err != nil {
+		t.Fatalf("Failed to process environment variable: %v.", err)
+	}
+
+	t.Parallel()
+	ctx := context.Background()
+
+	clients := e2e.SetupWithNamespace(t, test.TLSNamespace)
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "runtime",
+	}
+
+	if len(env.TLSServiceName) != 0 {
+		names.Service = env.TLSServiceName + "-dmtls"
+	}
+
+	// Clean up on test failure or interrupt.
+	test.EnsureTearDown(t, clients, &names)
+
+	// Set up initial Service.
+	svc, err := v1test.CreateServiceReady(t, clients, &names,
+		func(service *v1.Service) {
+			service.Annotations = map[string]string{"networking.knative.dev/disableAutoTLS": "True"}
+		})
+	if err != nil {
+		t.Fatalf("Failed to create initial Service %v: %v", names.Service, err)
+	}
+
+	// Using fixed hostnames can lead to conflicts when multiple tests run at
+	// once, so include the svc name to avoid collisions.
+	suffix := "example.com"
+
+	if env.CustomDomainSuffix != "" {
+		suffix = env.CustomDomainSuffix
+	}
+
+	host := "dm." + suffix
+
+	// Point DomainMapping at our service.
+	var dm *v1alpha1.DomainMapping
+	if err := reconciler.RetryTestErrors(func(int) error {
+		dm, err = clients.ServingAlphaClient.DomainMappings.Create(ctx, &v1alpha1.DomainMapping{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      host,
+				Namespace: svc.Service.Namespace,
+			},
+			Spec: v1alpha1.DomainMappingSpec{
+				Ref: duckv1.KReference{
+					Namespace:  svc.Service.Namespace,
+					Name:       svc.Service.Name,
+					APIVersion: "serving.knative.dev/v1",
+					Kind:       "Service",
+				},
+			},
+		}, metav1.CreateOptions{})
+		return err
+	}); err != nil {
+		t.Fatalf("Create(DomainMapping) = %v, expected no error", err)
+	}
+
+	t.Cleanup(func() {
+		clients.ServingAlphaClient.DomainMappings.Delete(ctx, dm.Name, metav1.DeleteOptions{})
+	})
+
+	// Wait for DomainMapping to go Ready.
+	if waitErr := wait.PollImmediate(test.PollInterval, test.PollTimeout, func() (bool, error) {
+		state, err := clients.ServingAlphaClient.DomainMappings.Get(context.Background(), dm.Name, metav1.GetOptions{})
```

Comment:
```suggestion
		state, err := clients.ServingAlphaClient.DomainMappings.Get(ctx, dm.Name, metav1.GetOptions{})
```

I guess we can re-use `ctx`.